dbtmetabase/parsers/metrics.py

```diff
@@ -0,0 +1,482 @@
+    # Numeric functions
+    "abs": {
+        "displayName": "abs",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "expressions",
+    },
+    "floor": {
+        "displayName": "floor",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "expressions",
+    },
+    "ceil": {
+        "displayName": "ceil",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "expressions",
+    },
+    "round": {
+        "displayName": "round",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "expressions",
+    },
+    "sqrt": {
+        "displayName": "sqrt",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "advanced-math-expressions",
+    },
+    "power": {
+        "displayName": "power",
+        "type": "number",
+        "args": ["number", "number"],
+        "requiresFeature": "advanced-math-expressions",
+    },
+    "log": {
+        "displayName": "log",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "advanced-math-expressions",
+    },
+    "exp": {
+        "displayName": "exp",
+        "type": "number",
+        "args": ["number"],
+        "requiresFeature": "advanced-math-expressions",
+    },
+    # Boolean functions
+    "contains": {
+        "displayName": "contains",
+        "type": "boolean",
+        "args": ["string", "string"],
+    },
+    "starts-with": {
+        "displayName": "startsWith",
+        "type": "boolean",
+        "args": ["string", "string"],
+    },
+    "ends-with": {
+        "displayName": "endsWith",
+        "type": "boolean",
+        "args": ["string", "string"],
+    },
+    "between": {
+        "displayName": "between",
+        "type": "boolean",
+        "args": ["expression", "expression", "expression"],
+    },
+    "time-interval": {
+        "displayName": "interval",
+        "type": "boolean",
+        "args": ["expression", "number", "string"],
+    },
+    "is-null": {
+        "displayName": "isnull",
+        "type": "boolean",
+        "args": ["expression"],
+    },
+    "is-empty": {
+        "displayName": "isempty",
+        "type": "boolean",
+        "args": ["expression"],
+    },
+    # Other expression functions
+    "coalesce": {
+        "displayName": "coalesce",
+        "type": "expression",
+        "args": ["expression", "expression"],
+        "multiple": True,
+    },
+    "case": {
+        "displayName": "case",
+        "type": "expression",
+        "args": [
+            "expression",
+            "expression",
+        ],  # ideally we'd alternate boolean/expression
+        "multiple": True,
+    },
+    # boolean operators
+    "and": {"displayName": "AND", "type": "boolean", "args": ["boolean", "boolean"]},
+    "or": {"displayName": "OR", "type": "boolean", "args": ["boolean", "boolean"]},
```

Comment:
I noticed a bug in the compiler with the following expression: `metabase_compiler.transpile_expression("CountIf(isnull([startup_pitched_at] OR NOT isnull([startup_pitched_at])))")`
which compiles to `['count']`

It seems to be related to "NOT OR" because if I remove the NOT operator it works as expected
`metabase_compiler.transpile_expression("CountIf(isnull([startup_pitched_at] OR isnull([startup_pitched_at])))")`
=> `['count-where', ['is-null', ['or', ['field', 1417549, None], ['is-null', ['field', 1417549, None]]]]]`

or if I use parenthesis, `metabase_compiler.transpile_expression("CountIf(isnull([startup_pitched_at] OR (NOT isnull([startup_pitched_at]))))")`
=> `['count-where', ['is-null', ['or', ['field', 1417549, None], ['not', ['is-null', ['field', 1417549, None]]]]]]`

The expression at the top is valid in Metabase and was copy/pasted from the UI so I would hope it works in this compiler.

Reply:
Thanks for this @remigabillet ! I will take a look at the compiler during the break. Think we are ready to move this into a beta in the master branch with a little documentation, variable renaming, etc. Need to look more deeply into dbt's metrics definitions too are they push along their 1.0 rc's.